src/connection.cpp

```diff
@@ -154,11 +144,13 @@ void Connection::setMAC(std::string_view mac)
     if ((mac.size() != 12) && (mac.size() != 14) && (mac.size() != 17))
         throw std::runtime_error("Invalid MAC address size");
 
-    // TODO: Make it easier to read and consider the Cisco MAC identifier
-    const std::regex pattern("^([0-9A-Fa-f]{2}[:-]){5}"
-                             "([0-9A-Fa-f]{2})|([0-9a-"
-                             "fA-F]{4}\\.[0-9a-fA-F]"
-                             "{4}\\.[0-9a-fA-F]{4})$");
+    const std::regex pattern(
+        R"regex(^
+    ([0-9A-Fa-f]{2}[:-]){5}        # Matches MAC address with colons or hyphens
+    ([0-9A-Fa-f]{2})|              # Or
```

Comment:
Didn't get this one.

Reply:
What exactly? It's a pattern to verify if the MAC address is valid or not.

Reply:
> What exactly? It's a pattern to verify if the MAC address is valid or not.

I don't understand the documentation. what is the "Or"?

Reply:
The "or" stands for the operator "**|**"... Before that, "Matches MAC address with colons or hyphens", and after,  "Matches IPv6 format"

Reply:
Can we use the regex on the same line to further enhance the reading, I was confused due to that.

Reply:
> Can we use the regex on the same line to further enhance the reading, I was confused due to that.

https://github.com/viniciusferrao/cloysterhpc/pull/27/commits/1e9882e5b5a7d1ffd4beb8593f862c4d60ed7c11